nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/test/java/org/apache/nifi/atlas/provenance/analyzer/TestAwsS3DirectoryV2.java

```diff
@@ -76,17 +76,19 @@ protected void assertAnalysisResult(DataSetRefs refs, String dirPath) {
         Referenceable ref = refs.getOutputs().iterator().next();
 
         String actualPath = dirPath;
+        String parentPath = StringUtils.substringBeforeLast(actualPath, "/");
         while (StringUtils.isNotEmpty(actualPath) && !"/".equals(actualPath)) {
             String directory = StringUtils.substringAfterLast(actualPath, "/");
 
             assertEquals(AwsS3Directory.TYPE_DIRECTORY_V2, ref.getTypeName());
             assertEquals(String.format("s3a://%s%s/@%s", AWS_BUCKET, actualPath, ATLAS_NAMESPACE), ref.get(ATTR_QUALIFIED_NAME));
             assertEquals(directory, ref.get(ATTR_NAME));
-            assertEquals(actualPath + "/", ref.get(ATTR_OBJECT_PREFIX_V2));
+            assertEquals(parentPath + "/", ref.get(ATTR_OBJECT_PREFIX_V2));
```

Comment:
Minor simplification: `parentPath` could be calculated inline from `actualPath`:
```suggestion
            assertEquals(StringUtils.substringBeforeLast(actualPath, "/") + "/", ref.get(ATTR_OBJECT_PREFIX_V2));
```